### ISSUE TITLE
"command&" misparsed

### ISSUE THREAD

**Issue by astronouth7303** [xonsh/xonsh#2544]
`true&` is misparsed. It should execute `true` in the background, but instead treats the whole thing as the command name.

`true &` works fine.

**Comment by scopatz** [comment]
Thanks for reporting!

**Comment by scopatz** [comment]
I don't think that this is easily fixable, since one of the ways that the parser works is that it says whitespace counts in subprocess mode. Addressing this would special case backgrounding to have whitespace not matter for the last token. That said, we probably would still want this....

**Comment by corpix** [comment]
This also affects `bash -c 'sleep 5 &'` invocation. Example:

> `xonsh $` indicates a xonsh prompt
> `bash $` indicates a bash prompt

``` console
xonsh $ import time; start = time.time(); bash -c 'sleep 5 &'; print(time.time() - start)
5.01800537109375

bash $ time bash -c 'sleep 5 &'
real    0m0.014s
user    0m0.008s
sys     0m0.005s
```

This make tools that use background jobs unusable in the xonsh. For example `pass -c something` (https://www.passwordstore.org/) just blocks the shell(until password is not deleted from clipboard).

**Comment by scopatz** [comment]
Hi @corpix - I think bash is doing something very strange here.... I am not sure that is the correct behaviour to replicate.

**Comment by scopatz** [comment]
Though the pass case is a notable usecase.

**Comment by whitelynx** [comment]
I'm having similar behavior with `nvim-qt`; instead of backgrounding like it does in other shells, it blocks the `xonsh` session until I exit `nvim-qt`.

Seems like that's probably a separate issue from this one... if I don't find an existing issue for it, I'll open one.

**Comment by whitelynx** [comment]
Actually, it looks like my issue is a duplicate of #2060... I'll try the hack from that issue.
